File: 10º Semestre/Sistemas Distribuidos/Exp5/exercicio 2/java/Main.java
import java.util.Base64;
import demo.PersonProto.Person;

public class Main {
  public static void main(String[] args) {
    try {
        byte[] decoded = Base64.getDecoder().decode(args[0]);
        Person p = Person.parseFrom(decoded);
        System.out.printf("Name: %s\n", p.getName());
        System.out.printf("EnrollNumber: %d\n", p.getEnrollNumber());
        System.out.printf("Height: %.2f\n", p.getHeight());
        System.out.printf("LuckNumbers: %s%n", p.getLuckNumbersList());
    } catch (Exception e) {
        System.out.println(e);
    }
  }
}

File: 10º Semestre/Sistemas Distribuidos/Exp5/exercicio 2/python/main.py
from person_pb2 import Person
from base64 import b64encode

if __name__ == "__main__":
    p = Person()
    # Coloque seu Nome, RA, altura e escolha 3 números
    p.Name = "Nome de Alguem"
    p.EnrollNumber = 123456
    p.Height = 1.77
    p.LuckNumbers.extend([7, 23, 47])
    # serialize in python binary "string"
    bstr = p.SerializeToString()
    # transform into a normal string
    txt = b64encode(bstr).decode()
    # print the string
    print(txt);